sound/soc/sof/intel/hda-bus.c

```diff
@@ -41,6 +55,9 @@ static void sof_hda_bus_link_power(struct hdac_device *codec, bool enable)
 	 */
 	if (codec->addr == HDA_IDISP_ADDR && !enable)
 		snd_hdac_display_power(bus, HDA_CODEC_IDX_CONTROLLER, false);
+
+	/* WAKEEN needs to be set for disabled links */
+	update_codec_wake_enable(bus, codec->addr, enable);
```

Comment:
the change looks fine, what I am wondering is whether this path can be taken when going to S3, in which case we probably don't want to set those bits? The commit message says it's for pm_runtime only, isn't it?

Reply:
Hmm, true. This could leave WAKEEN enabled for S3 suspend. Let me add logic to clear WAKEEN when we go to S3.